apps/discovery_api/lib/discovery_api_web/plugs/set_current_user.ex

```diff
@@ -2,13 +2,67 @@ defmodule DiscoveryApiWeb.Plugs.SetCurrentUser do
   @moduledoc """
   Convenience plug to set current user on connection - basically aliasing loaded resource
   """
+  use Properties, otp_app: :discovery_api
 
+  getter(:raptor_url, generic: true)
   import Plug.Conn
-
+  alias DiscoveryApi.Schemas.Users
   def init(default), do: default
 
   def call(conn, _) do
     current_user = Guardian.Plug.current_resource(conn)
+
+    if System.get_env("REQUIRE_API_KEY") == "true" do
+      assign_current_user(conn, current_user, get_api_key_from_header(conn))
+    else
+      assign(conn, :current_user, current_user)
+    end
+  end
+
+  defp assign_current_user(conn, current_user, api_key) when not is_nil(current_user) do
     assign(conn, :current_user, current_user)
   end
+
+  defp assign_current_user(conn, current_user, api_key) when is_nil(current_user) and is_nil(api_key) do
+    render_401_missing_api_key(conn)
+    #    assign(conn, :current_user, current_user)
```

Comment:
testing a few things around this 401 missing case, will remove comment after that's done